src/aggregator/aggregator/generic_elem.go

```diff
@@ -345,14 +345,15 @@ func (e *GenericElem) expireValuesWithLock(
 
 func (e *GenericElem) expireFlushState() {
 	for _, t := range e.flushStateToExpire {
-		prevFlushState, ok := e.flushState[t]
+		fState, ok := e.flushState[t]
 		if !ok {
 			ts := t.ToTime()
 			instrument.EmitAndLogInvariantViolation(e.opts.InstrumentOptions(), func(l *zap.Logger) {
 				l.Error("expire time not in state map", zap.Time("ts", ts))
 			})
+			return
```

Comment:
I wonder if it'd be safer to `continue` so a bad val doesn't prevent all expirations

Reply:
ugh. good call.